onnxruntime/core/providers/rocm/math/matmul.cc

```diff
@@ -110,9 +118,9 @@ Status MatMul<T>::ComputeInternal(OpKernelContext* ctx) const {
 
   rocblas_operation transA = transa ? rocblas_operation_transpose : rocblas_operation_none;
   rocblas_operation transB = transb ? rocblas_operation_transpose : rocblas_operation_none;
-  const int lda = transa ? static_cast<int>(helper.M()) : static_cast<int>(helper.K());
-  const int ldb = transb ? static_cast<int>(helper.K()) : static_cast<int>(helper.N());
-  const int ldc = static_cast<int>(helper.N());
+  const int lda = helper.Lda(transa);
+  const int ldb = helper.Ldb(transb);
+  const int ldc = helper.Ldc();
   int64_t stride_A, stride_B, stride_C, batch_count;
 
   if (helper.OutputOffsets().size() == 1) {
```

Comment:
does this branch need "helper.FillOffsets();"?

Reply:
No need. The fillOffsets is to populate left_offsets_, right_offsets_ and output_offsets_. These are not used in this branch.